es_cli/cli.py

```diff
@@ -239,13 +239,17 @@ def remap(mapping, index_url):
         index=orig_index
     ).get(orig_index, {}).get('aliases', {}).keys()
 
-    mappings = {}
+    index_body = {'mappings': {}}
     for mapping_fname in mapping:
         with open(mapping_fname) as mapping_fd:
             mapping_dict = json.loads(mapping_fd.read())
-            mappings.update(mapping_dict['mappings'])
+            for key, val in mapping_dict.items():
+                if key not in index_body:
+                    index_body[key] = {}
 
-    body = json.dumps({'mappings': mappings})
+                index_body[key].update(mapping_dict[key])
```

Comment:
not at that level, but at the next, analyzers, for example... that is enough for the mappings, as they should have different names, but for setting is not enough, you are right...